back/spec/services/permissions_service_spec.rb

```diff
@@ -188,25 +197,39 @@
       end
 
       context 'when fully registered unconfirmed resident' do
-        before { user.update!(email_confirmed_at: nil) }
+        before { user.reset_confirmation_and_counts }
 
-        it { expect(denied_reason).to be_nil }
+        it { expect(denied_reason).to eq 'missing_data' }
       end
 
       context 'when fully registered confirmed resident' do
         it { expect(denied_reason).to be_nil }
       end
 
+      context 'when fully registered sso user' do
+        before do
+          facebook_identity = create(:facebook_identity)
+          user.identities << facebook_identity
+          user.update(password_digest: nil)
```

Comment:
`update` can fail silently, which can make debugging a possible failure hard. I would suggest to either use `user.password_digest = nil` or `update!`.